--- trunk/catalog/model/catalog/product.php (revision 358)
+++ trunk/catalog/model/catalog/product.php (revision 374)
@@ -54,6 +54,7 @@
 					$implode = array();
 					
 					$words = explode(' ', $data['filter_name']);
+					$words = array_filter(array_unique($words), 'strlen');
 					
 					foreach ($words as $word) {
 						if (!empty($data['filter_description'])) {
@@ -76,6 +77,7 @@
 					$implode = array();
 					
 					$words = explode(' ', $data['filter_tag']);
+					$words = array_filter(array_unique($words), 'strlen');
 					
 					foreach ($words as $word) {
 						$implode[] = "LCASE(pt.tag) LIKE '%" . $this->db->escape(utf8_strtolower($word)) . "%' AND pt.language_id = '" . (int)$this->config->get('config_language_id') . "'";
@@ -430,6 +432,7 @@
 				$implode = array();
 				
 				$words = explode(' ', $data['filter_name']);
+				$words = array_filter(array_unique($words), 'strlen');
 				
 				foreach ($words as $word) {
 					if (!empty($data['filter_description'])) {
@@ -452,6 +455,7 @@
 				$implode = array();
 				
 				$words = explode(' ', $data['filter_tag']);
+				$words = array_filter(array_unique($words), 'strlen');
 				
 				foreach ($words as $word) {
 					$implode[] = "LCASE(pt.tag) LIKE '%" . $this->db->escape(utf8_strtolower($word)) . "%' AND pt.language_id = '" . (int)$this->config->get('config_language_id') . "'";

